src/averageAges.js

```diff
@@ -55,10 +73,18 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // 1. find a mother of each person (or only for men)
-  // 2. keep people who have mothers in the array
-  // 3. calculate the difference child.born - mother.born
-  // 4. return the average value
+  const children = people
+    .filter(({ sex, mother }) => !onlyWithSon
```

Comment:
Я теж ламав голову як скоротити цю ділянку коду) Ми тут можемо першочергово знайти людину зі ім'ям потім перевіряти чи присутнє onlyWithSon, в кейсі коли присутнє додавати умову, якщо ж ні, друга умова завжди буде тру
```suggestion
    .filter(({ sex, mother }) => people.some(({ name }) => name = mother) && (onlyWithSon ? sex = m : true);
```
Ну і в такому ключі на інших ділянках коду.